include/lbann/callbacks/callback_checksmall.hpp

```diff
@@ -49,7 +49,7 @@ class lbann_callback_checksmall : public lbann_callback {
   lbann_callback_checksmall() : lbann_callback() {}
   lbann_callback_checksmall(const lbann_callback_checksmall&) = default;
   lbann_callback_checksmall& operator=(
-    const lbann_callback_checksmall&) = default;
+    const lbann_callback_checksmall&) = delete;
```

Comment:
I'd prefer making `m_threshold` a `static const` or `static constexpr` variable. That way we can keep the copy-assignment.